File: 01Strings/00ispanagram.js
let TOTAL = 0;

main();


function main() {
    let string = "abcdefghijklmnopqrstuvwxyz";
    sumASCII();
    console.log(isPanagram(string));
}

function sumASCII() {
    let auxsum = 0;
    let aux = ["a", "z"];
    let x_i = aux[0].charCodeAt(0);
    let x_f = aux[1].charCodeAt(0);
    
    for ( let i = x_i; i <= x_f; i++ )
        auxsum += i;
    
    TOTAL = auxsum;
    return;
}

function isPanagram(string) {
    let acum = 0;
    let letters = {};
    for ( let char of string ){
        if ( !letters[char] ){
            letters[char] = char.charCodeAt(0);
            acum += char.charCodeAt(0);
        }
    }
    return acum == TOTAL;
}

File: 01Strings/00ispanagram.py
TOTAL = 0

def sumASCII() -> None:
    global TOTAL
    x_i = ord("a")
    x_f = ord("z")
    for i in range(x_i, x_f+1):
        TOTAL += i
    return

def isPanagram( string: str ) -> bool:
    letters = {}
    counter = 0
    for char in string:
        if char not in letters:
            letters[char] = ord(char)
            counter += ord(char)
    return counter == TOTAL

if __name__ == "__main__":
    string = "abcdefghijklmnopqrstuvwxyz"
    sumASCII()
    print(isPanagram(string))